File: 3507. Minimum Pair Removal to Sort Array I/3507.py
class Solution:
    def minimumPairRemoval(self, nums: list[int]) -> int:

        numberOperation = 0
        isDecreasing = True

        while isDecreasing:
            if len(nums) == 1:
                return numberOperation
            
            isDecreasing = False

            minSum = [0,float('inf')]

            for i in range(len(nums)-1):
                if nums[i] > nums[i+1]:
                    isDecreasing = True

                sumAd = nums[i] + nums[i+1]

                if sumAd < minSum[1]:
                    minSum[0] = i
                    minSum[1] = sumAd
            
            if isDecreasing == True:
                del nums[minSum[0]+1]
                nums[minSum[0]] = minSum[1]
                numberOperation += 1
            print(nums)

        return numberOperation
    

S = Solution()
print(S.minimumPairRemoval([689,-360,234,673,663,-741,480,860,-707,209,246,792,930,696,-305]))


        
        

File: 3507. Minimum Pair Removal to Sort Array I/3507.cs
public class Solution {
    public int MinimumPairRemoval(int[] nums) {
        List<int> lista = new List<int>(nums);
        int operacoes = 0;

        while (true) {
            if (lista.Count <= 1){
                return operacoes;
            }

            bool estaOrdenado = true;
            long menorSoma = long.MaxValue; 
            int minIdx = -1;

            for (int i = 0; i < lista.Count - 1; i++) {
                if (lista[i] > lista[i + 1]) {
                    estaOrdenado = false;
                }
                long somaAtual = (long)lista[i] + lista[i + 1];

                if (somaAtual < menorSoma) {
                    menorSoma = somaAtual;
                    minIdx = i;
                }
            }

            if (estaOrdenado) {
                return operacoes;
            }

            lista[minIdx] = (int)menorSoma;
            lista.RemoveAt(minIdx + 1);
            operacoes++;
        }
    }
}